packages/apps/shopify-api/lib/billing/__tests__/create-usage-record.test.ts

```diff
@@ -0,0 +1,178 @@
+import {queueMockResponses} from '../../__tests__/test-helper';
+import {testConfig} from '../../__tests__/test-config';
+import {Session} from '../../session/session';
+import {LATEST_API_VERSION} from '../../types';
+import {
+  shopifyApi,
+  BillingError,
+  BillingInterval,
+  BillingConfigUsageLineItem,
+  BillingConfigSubscriptionLineItemPlan,
+} from '../..';
+
+import * as Responses from './responses';
+
+const DOMAIN = 'test-shop.myshopify.io';
+const ACCESS_TOKEN = 'access-token';
+const GRAPHQL_BASE_REQUEST = {
+  method: 'POST',
+  domain: DOMAIN,
+  path: `/admin/api/${LATEST_API_VERSION}/graphql.json`,
+  headers: {'X-Shopify-Access-Token': ACCESS_TOKEN},
+};
+
+describe('shopify.billing.createUsageRecord', () => {
+  const session = new Session({
+    id: '1234',
+    shop: DOMAIN,
+    state: '1234',
+    isOnline: true,
+    accessToken: ACCESS_TOKEN,
+    scope: 'read_returns',
+  });
+
+  const billing = {
+    usage: {
+      lineItems: [
+        {
+          interval: BillingInterval.Usage,
+          amount: 5.0,
+          currencyCode: 'USD',
+          terms: '1 dollar per usage',
+        } as BillingConfigUsageLineItem,
+      ],
+    } as BillingConfigSubscriptionLineItemPlan,
+  };
+
+  test('returns the details of the usage record created, when passing the subscription line item ID', async () => {
+    const shopify = shopifyApi(testConfig({billing}));
+    queueMockResponses([Responses.USAGE_RECORD_CREATE_RESPONSE]);
+
+    const description = Responses.USAGE_RECORD_DESCRIPTION;
+    const price = Responses.USAGE_RECORD_PRICE;
+    const subscriptionId = Responses.USAGE_RECORD_SUBSCRIPTION_ID;
+    const response = await shopify.billing.createUsageRecord({
+      session,
+      description,
+      price,
+      subscriptionLineItemId: subscriptionId,
+    });
+
+    expect(response).toEqual(
+      JSON.parse(Responses.USAGE_RECORD_CREATE_RESPONSE).data
+        .appUsageRecordCreate.appUsageRecord,
```

Comment:
Nit: if we'll be using this response to test more than a couple of tests, we could consider exporting it as a separate const to make these `expect` calls a bit easier to read.